ArcGISRuntimeSDKQt_CppSamples/Routing/OfflineRouting/OfflineRouting.cpp

```diff
@@ -0,0 +1,271 @@
+}
+
+int OfflineRouting::travelModeIndex()
+{
+  return m_travelModeIndex;
+}
+
+void OfflineRouting::connectSignals()
+{
+  connect(m_routeTask, &RouteTask::createDefaultParametersCompleted, this, [this](QUuid, const RouteParameters& defaultParameters){
+    m_routeParameters = defaultParameters;
+  });
+
+  connect(m_routeTask, &RouteTask::doneLoading, this, [this](Error loadError){
+    if (loadError.isEmpty())
+    {
+      m_routeTask->createDefaultParameters();
+      emit travelModeNamesChanged();
+    }
+    else
+    {
+      qDebug() << loadError.message() << loadError.additionalMessage();
+      return;
+    }
+  });
+
+  connect(m_mapView, &MapQuickView::identifyGraphicsOverlayCompleted, this, [this](QUuid, IdentifyGraphicsOverlayResult* rawIdentifyResult)
+  {
+    QScopedPointer<IdentifyGraphicsOverlayResult> result(rawIdentifyResult);
+    if (!result->error().isEmpty())
+      qDebug() << result->error().message() << result->error().additionalMessage();
+
+    if (result->graphics().isEmpty())
+    {
+      m_selectedGraphic = nullptr;
+    }
+    // identify selected graphic in m_stopsOverlay
+    else
+    {
+      m_selectedGraphic = nullptr;
+      // find corresponding graphic in m_stopsOverlay
+      for (int i = 0; i < m_stopsOverlay->graphics()->size(); ++i)
+      {
+        if (m_stopsOverlay->graphics()->at(i)->operator==(result->graphics().at(0)))
+        {
+          m_selectedGraphic = m_stopsOverlay->graphics()->at(i);
+          break;
+        }
+      }
+    }
+
+  });
+
+  // check whether mouse pressed over an existing stop
+  connect(m_mapView, &MapQuickView::mousePressed, this, [this](QMouseEvent& e){
+    m_mapView->identifyGraphicsOverlay(m_stopsOverlay, e.x(), e.y(), 10, false);
+  });
+
+  // get stops from clicked locations
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& e){
+    if (!m_selectedGraphic)
+    {
+      SimpleMarkerSymbol* stopLabel = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Circle, Qt::red, 20, this);
+      Graphic* stopGraphic = new Graphic(m_mapView->screenToLocation(e.x(), e.y()), stopLabel, this);
+      m_stopsOverlay->graphics()->append(stopGraphic);
+      findRoute();
+    }
+    e.accept();
+  });
+
+  connect(m_mapView, &MapQuickView::mouseReleased, this, [this](QMouseEvent& e) {
+    if (m_selectedGraphic)
+    {
+      m_selectedGraphic = nullptr;
+      e.accept();
+    }
+  });
+
+  connect(m_mapView, &MapQuickView::mouseMoved, this, [this](QMouseEvent&e){
+    if (m_selectedGraphic)
+    {
+      e.accept();
+      m_selectedGraphic->setGeometry(m_mapView->screenToLocation(e.x(), e.y()));
+      findRoute();
+    }
+  });
+
+  connect(m_routeTask, &RouteTask::solveRouteCompleted, this, [this](QUuid, const RouteResult routeResult){
+    // clear old route
+    m_routeOverlay->graphics()->clear();
+    Polyline routeGeometry = routeResult.routes().first().routeGeometry();
+    Graphic* routeGraphic = new Graphic(routeGeometry, this);
```

Comment:
so, every route that is solved will add a new graphic? Rather than updating the old one?

Reply:
Yes, this is following the Java implementation of the sample